centreon/www/widgets/src/centreon-widget-resourcestable/src/Listing/Actions/Acknowledge/Dialog.tsx

```diff
@@ -0,0 +1,157 @@
+import useAclQuery from '../aclQuery';
+
+import { AcknowledgeFormValues } from '.';
+
+interface Props extends Pick<FormikHandlers, 'handleChange'> {
+  canConfirm: boolean;
+  errors?: FormikErrors<AcknowledgeFormValues>;
+  onCancel: () => void;
+  onConfirm: () => Promise<unknown>;
+  resources: Array<Resource>;
+  submitting: boolean;
+  values: FormikValues;
+}
+
+const useStyles = makeStyles()((theme) => ({
+  notify: {
+    marginBottom: theme.spacing(2)
+  }
+}));
+
+const DialogAcknowledge = ({
+  resources,
+  canConfirm,
+  onCancel,
+  onConfirm,
+  errors,
+  values,
+  submitting,
+  handleChange
+}: Props): JSX.Element => {
+  const { classes } = useStyles();
+
+  const { t } = useTranslation();
+
+  const { getAcknowledgementDeniedTypeAlert, canAcknowledgeServices } =
+    useAclQuery();
+
+  const deniedTypeAlert = getAcknowledgementDeniedTypeAlert(resources);
+
+  const open = resources.length > 0;
+
+  const hasHosts = resources.find((resource) => resource.type === 'host');
+
+  return (
+    <Dialog
+      confirmDisabled={!canConfirm}
+      data-testid="dialogAcknowledge"
+      labelCancel={t(labelCancel)}
+      labelConfirm={t(labelAcknowledge)}
+      labelTitle={t(labelAcknowledge)}
+      open={open}
+      submitting={submitting}
+      onCancel={onCancel}
+      onClose={onCancel}
+      onConfirm={onConfirm}
+    >
+      <Grid container direction="column">
+        {deniedTypeAlert && (
+          <Grid item>
+            <Alert severity="warning">{deniedTypeAlert}</Alert>
+          </Grid>
+        )}
+        <Grid item>
+          <TextField
+            fullWidth
+            multiline
+            dataTestId={labelComment}
+            error={errors?.comment}
+            label={t(labelComment)}
+            rows={3}
+            value={values.comment}
+            onChange={handleChange('comment')}
+          />
+        </Grid>
+        <Grid container item className={classes.notify}>
+          <Grid item>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  checked={values.notify}
+                  color="primary"
+                  inputProps={{ 'aria-label': t(labelNotify) }}
+                  size="small"
+                  onChange={handleChange('notify')}
+                />
+              }
+              label={t(labelNotify) as string}
+            />
+          </Grid>
+          <Grid container item rowSpacing={1}>
+            <FormHelperText>{t(labelNotifyHelpCaption)}</FormHelperText>
+          </Grid>
+        </Grid>
+        <Grid item>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={values.isSticky}
+                color="primary"
+                inputProps={{ 'aria-label': t(labelSticky) }}
+                size="small"
+                onChange={handleChange('isSticky')}
+              />
+            }
+            label={t(labelSticky) as string}
```

Comment:
Small question : why add ( as string ) I don't think it needed

Reply:
it's just to prevent typescript errors as the newest version of useTranslation returns a non-string value